combined-opt documentation, create dot resource file

milljm opened the pull request:
Adding documentation on how to use the MOOSE binary provided
by idaholab/public channel.

Add link for Reactor Meshing workshop

Adding relap5-3d manual link. We have yet to create a DNS entry
within the HPC enclave. So the link may not yet work during the
time of this PR.

Create dot resource file

Set umat test to only run if build is not of the installed type.

Closes #24049 #24165 #20729 #24195

Commit:
combined-opt documentation. Create dot resource

Adding documentation on how to use the MOOSE binary provided
by idaholab/public channel.

Add link for Reactor Meshing workshop

Adding relap5-3d manual link. We have yet to create a DNS entry
within the HPC enclave. So the link may not yet work during the
time of this PR.

Create dot resource file

Set umat test to only run if build is not of the installed type.

Closes #24049 #24165 #20729

milljm commented:
Using this PR to get umat plugin test copy to work (and documentation regarding moose-opt)

GiudGiud commented on modules/doc/config.yml:
```diff
@@ -41,6 +41,7 @@ Extensions:
         menu:
             Training:
                 MOOSE Framework (June): https://inlhrfedramp.gov1.qualtrics.com/jfe/form/SV_1RDZf244xsF1NK6
+                Reactor Meshing workhsop: https://forms.office.com/Pages/ResponsePage.aspx?id=haH8DPcl40mK53BNUybihfvSXpIhvVBAkfHDzM_V-pZUQlczTDczOFhCRU1MWVc0N1ZaNjhSQzM4Uy4u&wdLOR=cFBABC253-9278-411E-AE37-EB57D314024F
```
```suggestion
                Reactor Meshing workshop: https://forms.office.com/Pages/ResponsePage.aspx?id=haH8DPcl40mK53BNUybihfvSXpIhvVBAkfHDzM_V-pZUQlczTDczOFhCRU1MWVc0N1ZaNjhSQzM4Uy4u&wdLOR=cFBABC253-9278-411E-AE37-EB57D314024F
```

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1483644/) on fc7494b wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>e84297</th>
      <th colspan=4>#24167 fc7494</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.15%</td>
      <td>85.15%</td>
      <td>-0.00%</td>
      <td>0.00%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>86882</td>
      <td>86881</td>
      <td>-1</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15155</td>
      <td>15156</td>
      <td>+1</td>
      <td>1</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/24167/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/24167/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/24167/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/24167/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/24167/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/24167/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/24167/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/24167/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/24167/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/24167/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/24167/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/24167/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/24167/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/24167/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/24167/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/24167/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/24167/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/24167/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/24167/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/24167/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/24167/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/24167/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/24167/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/24167/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/24167/coverage/scalar_transport)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/24167/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/24167/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/24167/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/24167/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/24167/coverage/xfem)

</details>

## Warnings

- `framework` new line coverage rate 0.00% is less than the suggested 90.0%



This comment will be updated on new commits.

Commit:
Pagination, adding unittest

Add more pagination for combined-opt (moose-opt, moose, whatever),
Manual <compiler> Install, etc.

Add Unittest for skipped install routine.

Closes #24195

milljm commented on modules/doc/content/getting_started/new_users.md.template:
Query about googlegroups.


```
$MOOSE_DIR
```
...is being suggested, but not explained.

```
-j4
```
…is inconsistent with other instructions using -j6

pagination is linking back one page too far.


Modify clone_moose.md.template: put -> contain

GiudGiud replied:
we re not using announcements list afaik anymore

milljm commented:
While functional, our site can look way better on an iPad than it does. I am still seeing excessive left and right margins. The text is too small.

milljm commented on modules/doc/content/getting_started/installation/llvm_install_moose.md:
Whoops! Need to add build:
```
!template load file=installation/build_moose.md.template PATH=~/projects
```

milljm commented on modules/doc/content/getting_started/installation/install_moose.md:
Whoops! Need to add build:
```
!template load file=installation/build_moose.md.template PATH=~/projects
```

milljm commented on modules/doc/content/getting_started/installation/moose_conda_binary.md:
Add section on running tests:

```bash
moose --copy-inputs tests; cd moose/tests; moose --run -j6
```

milljm commented on modules/doc/content/getting_started/installation/moose_conda_binary.md:
```diff
@@ -0,0 +1,82 @@
+
+!style! halign=left
+Continue on to see more examples and tuturials using MOOSE! However, most of the next section is
+geared towards building your own application.
```
```diff
-geared towards building your application.
+geared towards developing your own application.
```

milljm reviewed (commented):
Fix make_install tests. Although, I think the logic is flawed in Tester.py now.

milljm committed:
Committing using github interface (cross fingers I am doing this right)

milljm committed:
Fix paginations, syntax, core count.

Commit:
Small work I was doing at home I want to keep

Trying to figure out why appending to yaml.dump does not
seem to work while in the same with open method...

Commit:
A working build

Changes to Makefiles representing a working build. But a bit hackish
and I think I need help.

TestHarness changes making use of the new 'install=true' feature.

Commit:
Add moose-dev Conda package

update tools, test-tools to not depend on moose-wasp. I am stealing
this from another PR this was happening in, just to get our
documentation up and rolling for upcoming workshops where we want to
use `mamba install moose` and `mamba install moose-dev`.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1499522/) on d71a055 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/tools

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

permcody commented on modules/combined/test/tests/umat/tests:
```diff
@@ -8,6 +8,7 @@
     exodiff = 'gap_heat_transfer_umat_out.e'
     library_mode = 'DYNAMIC'
     valgrind = 'NONE'
+    skip_installed = True
```
Haha - no way!
You really want to avoid these double negative combinations. I know it would be rare to only run a test in installed mode, but if we do that option would be `skip_installed = False`. 

Perhaps it should be an enum: `build_location = [IN_TREE, INSTALLED, BOTH]`?
The default would be `BOTH`.

milljm replied:
👍 
The way `compiler =` works is what I will follow:
```pre
set(['ALL', 'IN_TREE', 'INSTALLED'])
```

permcody commented on test/tests/make_install/tests:
```diff
@@ -102,7 +102,7 @@
                  rm -rf ../../../make_install_test'
       prereq = 'test_copy_install/run'
-      installed = False
+      skip_installed = True
```
Ha - looks like you've already mulled this over in your head.

permcody commented on framework/app.mk:
```diff
@@ -40,6 +40,11 @@ ifneq ($(SUFFIX),)
+ifeq ($(MOOSE_SKIP_DOCS),)
+  DOCUMENTATION := yes
+else
+  DOCUMENTATION := no
```
I'd prefer not to introduce a "duplicate" variable. Can we just use the value of `$MOOSE_SKIP_DOCS` directly?

GiudGiud replied:
seems like this should be fixed?
```
ifeq ($(MOOSE_SKIP_DOCS),)
```
ok with a new issue here too

permcody replied:
Fix `$MOOSE_SKIP_DOCS`? We can standardize on `TRUE` and `FALSE` instead of trying to handling things like `YES`, `NO`, etc. That would be preferable to setting more variables that do the same thing.

permcody commented on framework/app.mk:
```diff
@@ -335,12 +340,17 @@ $(app_HEADER): $(app_HEADER_deps)
 $(app_LIB): curr_deps := $(depend_libs)
 $(app_LIB): curr_libs := $(depend_libs_flags)
-$(app_LIB): $(app_HEADER) $(app_plugin_deps) $(depend_libs) $(app_objects) $(ADDITIONAL_DEPEND_LIBS)
+$(app_LIB): $(app_HEADER) $(app_plugin_deps) $(depend_libs) $(app_objects) $(ADDITIONAL_DEPEND_LIBS) $(dot_resource)
```
Looks like you got rid of `$(dot_resource)`

milljm replied:
Huh! Okay, I am a little confused how its currently working!

permcody commented on framework/app.mk:
```diff
@@ -375,25 +385,25 @@ endif
-applibs := $(call uniq,$(applibs))
-ifeq ($(libmesh_static),yes)
-  ifneq (,$(findstring clang,$(CXX)))
+ifneq (,$(findstring clang,$(CXX)))
```
I'm really confused with what's going on in this section. First, do we have any raw compilers still compiling MOOSE? That seems... bad. Did something prompt all these changes?

loganharbour replied:
Is it not better to just fully dump something like
```
$CXX --version
```

permcody replied:
👍

permcody reviewed (commented):
Sorry for the second review - seems I don't have more than about 3.5 mins at a time where I can focus on a task. It think your Makefile changes look good with the noted exceptions.

Commit:
Better spec attribute nameing convention

Use a better self describing attribute name for test skipping based
on binary installation type.

Use the same attribute value paradigm as the rest of the test spec
attributes ('ALL', etc, etc).

Update moose run_tests recipe to handle new module copyable inputs.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1502666/) on 6512c40 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/tools

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

Commit:
alter make_install/tests

The spec attribute has changed.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1502866/) on 012f5e6 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/tools

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

Commit:
DEBUG mode...

Commit:
Fix umat test spec attribute name

Commit:
Add additional module skips

Skip framework and external_petsc_solver when running module tests.

Modify Training menu to navigate to a markdown page. We have more
options with this being a page rather than on the menu.

Fix moose-wasp version the moose package was depending on.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1504314/) on 762ca71 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/tools

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1504741/) on a526b79 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/tools

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1504794/) on 34a22e8 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/tools

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

Commit:
Documentation Changes to support pre-build moose

Modify pre-built moose usage instructions to include new reactor
workshop module.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1505784/) on 595eb3a wanted to post the following:

A change of the following file(s) triggered this check:

  conda/tools

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

Commit:
Documentation fixes

tailor the documentation for the upcoming workshop.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1506628/) on e3723c5 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/tools

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

Commit:
For testing, use combined-opt

There is an issue with calling `moose` during testing. Switch back
to `combined-opt` for now.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1508582/) on 4b7edf4 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/tools

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

Commit:
Strange 'tests' vs 'no tests' directory

When operating on Darwin machines, the 'tests' directory is not needed.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1508813/) on c5fa0fa wanted to post the following:

A change of the following file(s) triggered this check:

  conda/tools

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

Commit:
Fix Unittests

I neglected to realize we are using special formating for Civet testing:

MOOSE_TERM_FORMAT="tpnsc"

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1509197/) on 61749aa wanted to post the following:

A change of the following file(s) triggered this check:

  conda/tools

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

Commit:
Do not test Intel Mac

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1509866/) on 27fc370 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/tools

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

Commit:
allow reactor_tutorials as copyable

`make install` to copy modules/tutorials a reactor_tutorials.
Set MOOSE_BIN environment (helps Workbench)

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1510288/) on 6651add wanted to post the following:

A change of the following file(s) triggered this check:

  conda/tools

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

Commit:
Hack to get moose, moose-opt --copy-inputs working

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1510453/) on e54b9a0 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/tools

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

Commit:
fix path mistake

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1510561/) on cfa3668 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/tools

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

Commit:
reactor_tutorial has a failing test

Skip testing reactor_tutorial, as it has a failing test: adv_examples.quarter_circle_connect

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1511155/) on 659a021 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/tools

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

Commit:
Use dict.update() built-in

Earlier Python versions do not support {**dict | dict} operands
(which I probably shouldn't have used in the first place). Using update
is easier to read.

GiudGiud commented on conda/moose/run_test.sh:
```diff
@@ -1,7 +1,47 @@
+fi
+
+# A list of module(s)/directories that we do not want to include for testing
+NOT_RUNNABLE=(doc module_load combined geochemistry misc navier_stokes tensor_mechanics framework external_petsc_solver reactor_tutorial)
```
this should be captured in an issue if it has not already

loganharbour replied:
https://github.com/idaholab/moose/issues/16857

loganharbour reviewed (commented):
Resource contents looks good to me for now. Looks extendable and is a great start with nothing that'll bite us later

permcody reviewed (dismissed):
I have checked over the Make changes and TestHarness changes and am satisfied. There are a few things that I think we can work on later, but I don't see any showstoppers from getting this in so we can work on our dry run for training. Thanks Jason!

GiudGiud commented on framework/app.mk:
```diff
@@ -495,11 +521,20 @@ else
+	@$(eval libpaths := $(foreach lib,$(applibs),$(dir $(lib))$(shell grep "dlname='.*'" $(lib) 2>/dev/null | sed -E "s/dlname='(.*)'/\1/g")))
+	@for lib in $(libpaths); do $(call patch_relink,$@,$$lib,$$(basename $$lib)); done
+
+ifeq ($(want_exec),yes)
+install_bin: $(bindst)
+else
+install_bin:
+endif
```
this is already defined higher?

GiudGiud commented on modules/doc/content/getting_started/installation/inl_hpc_prebuilt_moose.md:
````diff
@@ -29,28 +23,73 @@ module load use.moose moose-apps moose
+```
+
+You will see some information scroll by, and ultimately end back up at your prompt. If you perform a
+direcotry listing you should also see that an exodus file was generated in the process
````
```suggestion
directory listing you should also see that an exodus file was generated in the process
```

GiudGiud commented on modules/doc/content/getting_started/installation/inl_hpc_prebuilt_moose.md:
````diff
@@ -29,28 +23,73 @@ module load use.moose moose-apps moose
 
-```pre
-ex01_out.e
+For now, lets copy the reactor module into a safe location for editing:
````
```suggestion
For now, lets copy the reactor module workshop into a safe location for editing:
```

GiudGiud commented on modules/doc/content/getting_started/installation/moose_conda_binary.md:
````diff
@@ -0,0 +1,153 @@
+instruct `moose` to run the tests:
+
+```bash
+cd combined/reactor_workshop
````
it s odd to be using combined/reactor_workshop

why the 'combined' folder?

loganharbour replied:
Copied inputs go into `<binary name>/<installed name>`. This is a product of this process.

GiudGiud commented on modules/doc/content/getting_started/installation/moose_conda_binary.md:
````diff
@@ -0,0 +1,153 @@
+
+```bash
+cd ~/projects/examples/combined/reactor_workshop/tests/reactor_examples/abtr/
+moose -i abtr.i --mesh-only --error-deprecated
````
would remove the error-deprecated

seems bait for a live failure

GiudGiud commented on modules/doc/content/getting_started/installation/inl_hpc_prebuilt_moose.md:
````diff
@@ -29,28 +23,73 @@ module load use.moose moose-apps moose
+
+```bash
+cd ~/projects/examples/combined/reactor_workshop/tests/reactor_examples/abtr/
+moose -i abtr.i --mesh-only --error-deprecated
````
```suggestion
moose -i abtr.i --mesh-only
```

GiudGiud commented on modules/doc/content/getting_started/installation/moose_conda_binary.md:
````diff
@@ -0,0 +1,153 @@
+```
+
+You will see some information scroll by, and ultimately end back up at your prompt. If you perform a
+direcotry listing you should also see that an exodus file was generated in the process
````
```suggestion
directory listing you should also see that an exodus file was generated in the process
```

GiudGiud commented on modules/doc/content/getting_started/installation/moose_conda_binary.md:
````diff
@@ -0,0 +1,153 @@
+new `paraview` environment:
+
+```bash
+mamba activate base # just in case you have `moose` auto-activating
````
we can create from the moose environment

loganharbour replied:
Conda is dirty switching from environment to environment. it's better to do start in the base environment.

GiudGiud commented on modules/doc/content/getting_started/installation/moose_conda_binary.md:
````diff
@@ -0,0 +1,153 @@
+```
+
+## Viewing Results
+
````
we need a note that we must have a X window or something to do this?

loganharbour replied:
Nope

GiudGiud commented on modules/doc/content/getting_started/installation/true_update_conda.md:
```diff
@@ -0,0 +1,10 @@
+
+!alert note title=Always Update MOOSE and Conda together
+There is a tight dependency between libraries being provided by Conda, and what libraries MOOSE
+depends on. Therefor, when you update one you should always update the other.
```
```suggestion
depends on. Therefore, when you update one you should always update the other.
```

GiudGiud commented on modules/doc/content/getting_started/installation/true_update_conda.md:
````diff
@@ -0,0 +1,10 @@
+mamba update --all
+```
+
+!alert note title=Always Update MOOSE and Conda together
````
```suggestion
!alert note title=Always Update MOOSE and the Conda/Mamba packages together
```

GiudGiud commented on modules/doc/content/training/index.md:
```diff
@@ -0,0 +1,10 @@
+
+
+- [MOOSE Framework (2023 June 6-8, Idaho Falls)](https://inlhrfedramp.gov1.qualtrics.com/jfe/form/SV_1RDZf244xsF1NK6)
+- [Reactor Meshing Workshop (2023 May 16)](https://forms.office.com/Pages/ResponsePage.aspx?id=haH8DPcl40mK53BNUybihfvSXpIhvVBAkfHDzM_V-pZUQlczTDczOFhCRU1MWVc0N1ZaNjhSQzM4Uy4u&wdLOR=cFBABC253-9278-411E-AE37-EB57D314024F)
```
```suggestion
- [Reactor Meshing Workshop (2023 May 16, virtual)](https://forms.office.com/Pages/ResponsePage.aspx?id=haH8DPcl40mK53BNUybihfvSXpIhvVBAkfHDzM_V-pZUQlczTDczOFhCRU1MWVc0N1ZaNjhSQzM4Uy4u&wdLOR=cFBABC253-9278-411E-AE37-EB57D314024F)
```

GiudGiud commented on modules/doc/content/training/index.md:
```diff
@@ -0,0 +1,10 @@
+!style-end!
+
+
+- [MOOSE Framework (2023 June 6-8, Idaho Falls)](https://inlhrfedramp.gov1.qualtrics.com/jfe/form/SV_1RDZf244xsF1NK6)
```
```suggestion
- [MOOSE Framework (2023 June 6-8, Idaho Falls & Virtual)](https://inlhrfedramp.gov1.qualtrics.com/jfe/form/SV_1RDZf244xsF1NK6)
```

GiudGiud commented on python/TestHarness/util.py:
```diff
@@ -724,31 +725,13 @@ def getInitializedSubmodules(root_dir):
-    - If somebody installs tests into their git repository, this report an incorrect results
-
-    Neither of these cases a significant risk.
+    Read resource file and determin if binary was relocated
```
```suggestion
    Read resource file and determine if binary was relocated
```

loganharbour commented on conda/moose/build.sh:
```diff
@@ -12,11 +12,22 @@ make -j $CORES
 mkdir -p "${PREFIX}/etc/conda/activate.d" "${PREFIX}/etc/conda/deactivate.d"
 cat <<EOF > "${PREFIX}/etc/conda/activate.d/activate_${PKG_NAME}.sh"
 export PATH=\${PATH}:${PREFIX}/moose/bin
+export MOOSE_BIN=${PREFIX}/moose/bin/combined-opt
```
This isn't for beginners. And the file is in `bin`.

loganharbour committed:
Address review: grammar and minor doc changes

Co-authored-by: Guillaume Giudicelli <guillaume.giudicelli@gmail.com>

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1513020/) on 7fdfec1 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/tools

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

Commit:
Export all the variables, add relap5-discourse

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1513085/) on 1b180a9 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/tools

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

Commit:
Update instructions to be more general

Turns out Mac and Linux produce different `--copy-inputs` directories.
Re-word the instructions a bit to be more general.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1514769/) on 0de4a70 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/tools

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

milljm commented on conda/moose/build.sh:
```diff
@@ -26,11 +26,8 @@ mkdir -p "${PREFIX}/etc/conda/activate.d" "${PREFIX}/etc/conda/deactivate.d"
 export MOOSE_BIN=${PREFIX}/moose/bin/moose
-# The following will be deprecated
-export MOOSE_OPT=${PREFIX}/moose/bin/moose-opt
-export COMBINED_OPT=${PREFIX}/moose/bin/combined-opt
```
I decided against adding deprecated content...

Commit:
Template the workshop, Supply Windows instructs

Template the common "run an example" portion of our pre-built binaries

Supply a way for Windows users to take advantage of our pre-built binaries

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1517140/) on b1378dd wanted to post the following:

A change of the following file(s) triggered this check:

  conda/tools

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

Commit:
Updating versioner_hashes.yaml

As the final thing to do, bump versioner_hashes.yaml due to some
package changes.

Closes #24049 #24165 #20729 #24195

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1517522/) on 802cbf5 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/tools

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1517970/) on 3bdb8a6 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/tools

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

Commit:
Optionally load index (fix Griffin Docs)

Bump versioner_hashes.yaml

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1518964/) on b66fa8b wanted to post the following:

A change of the following file(s) triggered this check:

  conda/tools

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

milljm commented:
I've invalidated https://civet.inl.gov/job/1518997/ [No optional] once already for network timeouts. Ignore.
